### PR TITLE
feat(section): add with header bar variant

### PR DESCRIPTION
**Pull request by elviocb**
[SPC-1390]

### Description

Implement new variant **(withHeaderBar)** to `Section` component

Screen Shot 2021-11-04 at 19 27 18

### How to test

- Open Storybook and navigate to the `Variant` section in the `Section` component

### Review

- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/api-principles.md)
- [x] Annotate all `props` in component with documentation
- [x] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure that tests pass by running `yarn test`
- [x] Ensure that visuals tests pass by running `yarn test:visual`. If not - check the documentation [how to fix visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md#fixing-broken-visual-tests-inside-a-pr)
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

[SPC-1390]: https://toptal-core.atlassian.net/browse/SPC-1390?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by bogdanalexe90 — APPROVED** [review]

**Comment by bogdanalexe90** [comment]
@toptal-anvil ping reviewers

File: packages/picasso-lab/src/Section/styles.ts [hunk 1]
```diff
@@ -28,9 +28,28 @@ export default ({ sizes, palette }: Theme) =>
...
+      padding: 0,
+      '& > :last-child:not(:first-child)': {
+        padding: '1.5rem'
+      },
+      borderRadius: sizes.borderRadius.medium,
+      border: `solid ${sizes.borderWidth} ${palette.grey.light2}`
+    },
+    headerBar: {
+      display: 'flex',
+      padding: '0.75rem 1.063rem',
```
Comment (TomasSlama) [inline packages/picasso-lab/src/Section/styles.ts:41]: `17px` seems weird to me, most of the time designers are trying to have spacing dividable by 4px. Isn't it meant to be 16px + 1px border? I would double-check with designers

**Review by TomasSlama — COMMENTED** [review]
Looks good overall, can you please just check my comment?

**Review by workitponiedzialek — APPROVED** [review]
Looks like on design, but I agree that 17 is weird :smile:

File: packages/picasso-lab/src/Section/styles.ts (hunk 1 above)
Comment (elviocb) [reply packages/picasso-lab/src/Section/styles.ts:41]: I agree it's weird, I will change it to 1rem. Just to make sure,  I posted a question in the design channel

Comment (elviocb) [reply packages/picasso-lab/src/Section/styles.ts]: I confirmed with the design team, it's 16 😄

**Review by TomasSlama — APPROVED** [review]

File: packages/picasso-lab/src/Section/Section.tsx [hunk 2]
```diff
@@ -136,8 +137,12 @@ export const Section = forwardRef<HTMLDivElement, Props>(function Section(
...
         <Container
           data-testid={testIds?.header}
-          className={cx(classes.header, {
-            [classes.collapsedHeader]: collapsed
+          className={cx({
+            [classes.header]: variant !== 'withHeaderBar',
+            [classes.headerBar]: variant === 'withHeaderBar',
+            [classes.collapsedHeader]: variant !== 'withHeaderBar' && collapsed,
+            [classes.collapsedHeaderBar]:
+              variant === 'withHeaderBar' && collapsed
```
Comment (dmaklygin1) [inline packages/picasso-lab/src/Section/Section.tsx:145]:
```suggestion
            [classes[`header${variant}`]]: true,
            [classes[`collapsedHeader${variant}`]]: collapsed,
```
wdyt, won't it be so confusing then?

File: packages/picasso-lab/src/Section/Section.tsx [hunk 3]
```diff
@@ -136,8 +136,9 @@ export const Section = forwardRef<HTMLDivElement, Props>(function Section(
       {hasHeader && (
         <Container
           data-testid={testIds?.header}
-          className={cx(classes.header, {
-            [classes.collapsedHeader]: collapsed
+          className={cx({
+            [classes[`header${capitalize(variant)}`]]: true,
+            [classes[`collapsedHeader${capitalize(variant)}`]]: collapsed
```
Comment (dmaklygin1) [inline packages/picasso-lab/src/Section/Section.tsx:141]: wdyt, what if we place `variant` on the first place, so we may avoid usage of `capitalize`:
```suggestion
            [classes[`${variant}Header`]]: true,
            [classes[`${variant}CollapsedHeader`]]: collapsed
```

**Review by dmaklygin1 — APPROVED** [review]

**Pull request merged**